File: app/src/main/java/bassamalim/hidaya/other/AppDatabase.java
package bassamalim.hidaya.other;

import androidx.room.AutoMigration;
import androidx.room.Database;
import androidx.room.RoomDatabase;

import bassamalim.hidaya.interfaces.AthkarCategoryDao;
import bassamalim.hidaya.interfaces.AthkarDao;
import bassamalim.hidaya.interfaces.AyahDao;
import bassamalim.hidaya.interfaces.AyatRecitersDao;
import bassamalim.hidaya.interfaces.AyatTelawaDao;
import bassamalim.hidaya.interfaces.SuraDao;
import bassamalim.hidaya.interfaces.ThikrsDao;
import bassamalim.hidaya.models.AthkarCategoryDB;
import bassamalim.hidaya.models.AthkarDB;
import bassamalim.hidaya.models.AyatRecitersDB;
import bassamalim.hidaya.models.AyatTelawaDB;
import bassamalim.hidaya.models.JAyah;
import bassamalim.hidaya.models.SuraDB;
import bassamalim.hidaya.models.ThikrsDB;

@Database(
        entities = {JAyah.class, SuraDB.class, AthkarCategoryDB.class, AthkarDB.class,
        ThikrsDB.class, AyatRecitersDB.class, AyatTelawaDB.class},
        version = 1
        /*, autoMigrations = {@AutoMigration(from = 1, to = 2), @AutoMigration(from = 2, to = 3)}*/)
public abstract class AppDatabase extends RoomDatabase {

    public abstract AyahDao ayahDao();

    public abstract SuraDao suraDao();

    public abstract AthkarCategoryDao athkarCategoryDao();

    public abstract AthkarDao athkarDao();

    public abstract ThikrsDao thikrsDao();

    public abstract AyatRecitersDao ayatRecitersDao();

    public abstract AyatTelawaDao ayatTelawaDao();
}

File: app/src/main/java/bassamalim/hidaya/core/data/database/AppDatabase.kt
package bassamalim.hidaya.core.data.database

import androidx.room.Database
import androidx.room.RoomDatabase
import bassamalim.hidaya.core.data.database.daos.*
import bassamalim.hidaya.core.data.database.dbs.*

@Database(
    entities = [AyatDB::class, SuarDB::class, BooksDB::class,
        AthkarCategoryDB::class, AthkarDB::class, ThikrsDB::class,
        AyatRecitersDB::class, AyatTelawaDB::class, TelawatRecitersDB::class,
        TelawatVersionsDB::class, QuizQuestionsDB::class, QuizAnswersDB::class,
               CountryDB::class, CityDB::class],
    version = 1
)
abstract class AppDatabase : RoomDatabase() {
    abstract fun ayatDao(): AyatDao
    abstract fun suarDao(): SuarDao
    abstract fun booksDao(): BooksDao
    abstract fun athkarCategoryDao(): AthkarCategoryDao
    abstract fun athkarDao(): AthkarDao
    abstract fun thikrsDao(): ThikrsDao
    abstract fun ayatRecitersDao(): AyatRecitersDao
    abstract fun ayatTelawaDao(): AyatTelawaDao
    abstract fun telawatRecitersDao(): TelawatRecitersDao
    abstract fun telawatVersionsDao(): TelawatVersionsDao
    abstract fun telawatDao(): TelawatDao
    abstract fun quizQuestionDao(): QuizQuestionsDao
    abstract fun quizAnswerDao(): QuizAnswersDao
    abstract fun countryDao(): CountryDao
    abstract fun cityDao(): CityDao
}